packages/cubejs-cli/src/command/token.ts

```diff
@@ -0,0 +1,86 @@
+import chalk from 'chalk';
+import jwt from 'jsonwebtoken';
+import type { CommanderStatic } from 'commander';
+import { displayError, event, requireFromPackage } from '../utils';
+
+export const defaultExpiry = '30 days';
+
+const parsePayload = (payloadArray: string[] = []) => {
+  const result = {};
+
+  payloadArray.forEach((entry = '') => {
+    const [key, value] = entry.split('=');
+    if (key && value) result[key] = value;
+  });
+
+  return result;
+};
+
+type TokenOptions = {
+  expiry: string;
+  secret: string;
+  // Extra options
+  expiresIn?: string
+  payload: string[]
+  userContext: string[]
+};
+
+export const token = async (options: TokenOptions) => {
+  event('Generate Token');
+
+  const CubejsServer = await requireFromPackage('@cubejs-backend/server');
+  const { expiry = defaultExpiry, secret = CubejsServer.apiSecret() } = options;
```

Comment:
doesn't it mean that `expiry` and `secret` are optional?

Reply:
Good catch, fixed!